plugins/imagespacingbox/plugin.js

```diff
@@ -14,11 +14,89 @@
 						return;
 					}
 
+					var getFocusedWidget = function (editor) {
+						var widget = editor.widgets.focused;
+
+						if (widget && widget.name == 'image') return widget;
+
+						return null;
+					};
+
+					var getImageHVSpace = function (widget) {
+						var imageHVSpace = {hspace: '', vspace: ''};
+						var imageElement = widget.parts.image;
+						var style = imageElement.getAttribute('style');
```

Comment:
Hey @georgel-pop-lr,

Instead of using `getAttribute` here, don't you think we could access the `CSSStyleDeclaration` directly to avoid using regular expressions?

```js
var imageElement = widget.parts.image.$; // Get the native DOM element (not the `CKEDITOR.dom.element`)
var style = imageElement.style; // This should return a CSSStyleDeclaration 
```

Reply:
Hi @julien,

Thanks for reviewing, your idea is good, but I don't understand how we could remove the use of  regular expressions, because I use them to interpret and extract the value of the margin, in the style you could have others styles as well.

Regards.

Reply:
Well, if you have a `CSSStyleDeclaration` instance, you could use its `marginTop`, `marginRight`, `marginBottom` and `marginLeft` properties

Reply:
Didn't know about CSSStyleDeclaration, makes total, sense, changes pushed. Please check.

Regards.